airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -0,0 +1,797 @@
+         with the EKS Managed Nodegroup.
+    :type nodegroup_role_arn: str
+
+    """
+
+    template_fields = (
+        "cluster_name",
+        "cluster_role_arn",
+        "resources_vpc_config",
+        "nodegroup_name",
+        "nodegroup_role_arn",
+        "compute",
+        "aws_conn_id",
+        "region",
+    )
+
+    def __init__(
+        self,
+        cluster_name: str,
+        cluster_role_arn: str,
+        resources_vpc_config: Dict,
+        nodegroup_name: Optional[str] = None,
+        nodegroup_role_arn: Optional[str] = None,
+        compute: Optional[str] = DEFAULT_COMPUTE_TYPE,
+        aws_conn_id: Optional[str] = DEFAULT_CONN_ID,
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.cluster_name = cluster_name
+        self.cluster_role_arn = cluster_role_arn
+        self.resources_vpc_config = resources_vpc_config
+        self.compute = compute
+        self.aws_conn_id = aws_conn_id
+        self.region = region
+
+        if self.compute == 'nodegroup':
+            self.nodegroup_name = nodegroup_name or self.cluster_name + DEFAULT_NODEGROUP_NAME_SUFFIX
+            if nodegroup_role_arn:
+                self.nodegroup_role_arn = nodegroup_role_arn
+            else:
+                message = "Creating an EKS Managed Nodegroup requires nodegroup_role_arn to be passed in."
+                self.log.error(message)
+                raise AttributeError(message)
+
+    def execute(self, context):
+        eks_hook = EKSHook(
+            aws_conn_id=self.aws_conn_id,
+            region_name=self.region,
+        )
+
+        eks_hook.create_cluster(
+            name=self.cluster_name,
+            roleArn=self.cluster_role_arn,
+            resourcesVpcConfig=self.resources_vpc_config,
+        )
+
+        if self.compute is not None:
+            self.log.info("Waiting for EKS Cluster to provision.  This will take some time.")
+
+            countdown = TIMEOUT_SECONDS
+            while eks_hook.get_cluster_state(clusterName=self.cluster_name) != "ACTIVE":
+                if countdown >= CHECK_INTERVAL_SECONDS:
+                    countdown -= CHECK_INTERVAL_SECONDS
+                    self.log.info(
+                        "Waiting for cluster to start.  Checking again in %d seconds", CHECK_INTERVAL_SECONDS
+                    )
+                    sleep(CHECK_INTERVAL_SECONDS)
+                else:
+                    message = (
+                        "Cluster is still inactive after the allocated time limit.  "
+                        "Failed cluster will be torn down."
+                    )
+                    self.log.error(message)
+                    # If there is something preventing the cluster for activating, tear it down and abort.
+                    eks_hook.delete_cluster(name=self.cluster_name)
+                    raise RuntimeError(message)
```

Comment:
@ashb  - Working on removing the redundant catch/log/except blocks. How do you feel about this one? Should I drop this log as well, or is that reasonable?

Reply:
How long does deleting the cluster take? if it's more than a few seconds it's probably useful to keep both here, but on L300 the log and the exception are right next to each other, so we should remove the log form there.

Reply:
It's in the 15-20 minute range, in my experience.

Reply:
Oh one more thing: is `delete_cluster` sync or async? If it's async (and we just wait for the request) then we don't need the log.

Reply:
I thought I replied here but apparently not.  IKt's async, it sends the request to the API then moves on.  I've removed this logging in https://github.com/apache/airflow/pull/16571/commits/4d59785e903699d74d884a9137197baef3fe0fde